concert/app/src/functions/on_event_stage_changed/index.py

```diff
@@ -19,4 +35,36 @@ def handler(event, context):
         },
     )
 
+    song_list_update_response = update_song_list(message)
+
     return {"statusCode": 204}
+
+
+def update_song_list(message):
+    response = None
+
+    data = message["data"]
+    display_name = data.get(ATTR_DISPLAY_NAME)
+    start_time = data.get(ATTR_START_TIME)
+    end_time = data.get(ATTR_END_TIME)
+    if display_name and start_time and end_time:
+
+        song = [
+            {
+                ATTR_DISPLAY_NAME: display_name,
+                ATTR_START_TIME: start_time,
+                ATTR_END_TIME: end_time,
+            }
+        ]
+
+        update_args = {
+            "Key": {ATTR_RECORD_ID: RECORD_ID_SONG_LIST},
+            "UpdateExpression": "SET #songs = list_append(#songs, :song)",
+            "ExpressionAttributeNames": {"#songs": ATTR_SONGS},
+            "ExpressionAttributeValues": {":song": song},
+            "ReturnValues": "NONE",
+        }
+
+        response = table.update_item(**update_args)
```

Comment:
Since this issue builds off of #11, it makes sense to me to do all these updates in a single transaction. One network call.

Reply:
Done. Will have revision posted once I've finished updating tests.

I'm wondering, though: would it make more sense for the stageId write to be a `PutItem` rather than an `UpdateItem` request? Do we want attributes from old stages to linger?

Reply:
Yes, I think we would want a PutItem instead. Also, my suggestion about transactions might be a poor one considering my misunderstanding about updating the same record multiple times, as you pointed out.

Reply:
Since `SONG_LIST` and `EVENT_STAGE` are different records, and since each is only updated once per invocation of this handler, I think transactions should be fine here.

(Their syntax is really ugly, though, so I'd be totally fine not using them :wink: )